core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -767,6 +767,8 @@ object Chunk {
 
   /**
    * Returns a chunk backed by an array.
+   *
+   * WARNING: The array must not be mutated after creating the chunk.
```

Comment:
I wonder if it would make sense to have a naming convention to serve the same purpose as this warning. `Chunk.fromArray` could copy by default and then the alternate "unsafe" one would be the fast version. Depends if there is already a precedent for that in the ZIO codebase.

Also when seeing this, I also thought about if there might be a use for a `Chunk.fromArray[A](array: Array[A], start: Int, len: Int)` version.

Reply:
@reibitto I like the slicing version! Could be a nice PR if you have the cycles.

Re the naming: I think a warning is sufficient because we're not going to add the other variant. The caller can always defensively copy if required.